ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/WriteService.scala

```diff
@@ -95,12 +95,35 @@ trait WriteService
     * @param estimatedInterpretationCost Estimated cost of interpretation that may be used for
     *                                    handling submitted transactions differently.
     * @param telemetryContext            Implicit context for tracing.
-    * @return an async result of a SubmissionResult
     */
   def submitTransaction(
       submitterInfo: SubmitterInfo,
       transactionMeta: TransactionMeta,
       transaction: SubmittedTransaction,
       estimatedInterpretationCost: Long,
   )(implicit telemetryContext: TelemetryContext): CompletionStage[SubmissionResult]
+
+  /** Record the rejection of a command submission on the ledger.
+    *
+    * This method must be thread-safe.
+    *
+    * The result is communicated asynchronously via a [[ReadService]] implementation backed by the same participant
```

Comment:
```suggestion
    * This method is used by the Ledger API server to deliver interpretation failures as definitive answers to the
    * Ledger API client via completions; thereby allowing the Ledger API client to safely retry the command if desired.
    * The result is communicated asynchronously via a [[ReadService]] implementation backed by the same participant
```

Reply:
Added roughly. Note that the point of recording is that the application can decide to **not** retry. Retrying itself would always be safe thanks to the deduplication guarantees.